### PR TITLE
Move mask autoequip to a setting and remove actor mask interactions

### PR DESCRIPTION
**Pull request by mracsys**
Complement to #1992.

Moves the always-on mask autoequip hack to a cosmetic setting after community feedback.

Additionally, actor conversation branches when wearing a mask are removed if the setting is enabled. This allows collecting items from them without being forced to de-equip the mask, such as child rolling goron or lab dive as either age. This mostly addresses the softlock concerns of #1844, except for adult potentially being locked into fast bunny hood speed until the next time the player travels through time. #1992 handles the room restriction problems in #1844.

Removing the conversation branches is achieved by erasing the text IDs in the `sReactionTextIds` table for all actors except mask trading quest participants. Each actor has a unique reaction set ID based on a global search of references to `Text_GetFaceReaction`, making it easy to selectively tweak actor behavior without editing the actors themselves. Preserved actors are:
* en_heishi2 (Kakariko gate guard) - segment 0x05
* en_heishi4 (Other guards to be safe) - segments 0x06 and 0x07
* en_skj (skull kid) - segment 0x15
* en_cs (graveyard kid) - segment 0x0F
* en_mm (running man) - segment 0x1C

### CODE REVIEW

**Comment by r0bd0g** [comment]
When a setting drastically affects how the game plays like this, in no universe should it be a cosmetic option. Please, reserve cosmetic options for things that are actually cosmetic. If you must insist that two people could play the same seed with such drastically different gameplay, I think you will need to no longer refer to this group of settings as "cosmetics" throughout the randomizer.

I completely disagree with removing mask interactions. You're supposed to be able to talk to people and get their thoughts on masks. I don't think features should just be completely stripped from the game like this. (I remember some of the funnier text rando images being ironic mask reactions. Gameplay wise you could use masks to search for hints in full text shuffle, but that's probably not worth your time and probably neither is racing full text shuffle.) The only reason anybody ever talks to people with masks on is because they forgot to take the bunny hood off, which they wouldn't even be wearing, if not for the bunny hood speed up option that I've never really been a fan of either. You could just turn off bunny hood speed, and then none of this would be necessary. If you must insist -- please make it so that the mask interactions are only ignored when specifically the bunny hood is worn, and only when fast bunny hood is enabled? If this is intended to address softlock concerns, changing it like I suggest would still have this PR help some, but I must insist that the feature to keep masks equipped between scenes is either removed or properly fixed to not have masks stay permanently equipped as adult. (There's got to be a reasonable way to do that, right? Is there some reason we can't just unequip masks on savewarp?)

File: Patches.py [hunk 1]
```diff
@@ -1291,6 +1291,17 @@ def set_entrance_updates(entrances: Iterable[Entrance]) -> None:
     rom.write_int16(0x00E1F3CA, 0x5036)
     rom.write_int16(0x00E1F3CC, 0x5036)
 
+    # Remove mask reaction text IDs from all actors not involved in the
+    # mask trading sequence. See z_face_reaction.c in decomp for original
+    # values or https://wiki.cloudmodding.com/oot/Code_(File)/NTSC_1.0#Mask_Reaction_Text.
+    # Text ID 0x0000 is the default if a mask is not being worn.
+    # Convenience hack to allow the player to keep the bunny hood on when interacting
+    # with actors that give items (rolling goron, lab dive, etc.)
+    if world.settings.auto_equip_masks:
```
Comment (fenhl) [inline Patches.py]: I don't think this will work as intended, since it's checking the value of a cosmetic setting at patch creation time.

**Comment by mracsys** [comment]
> (There's got to be a reasonable way to do that, right? Is there some reason we can't just unequip masks on savewarp?)

The same function that's called to equip on scene change is also called after loading a save. I don't know how to distinguish those states. It's also a minor convenience feature to keep the bunny hood on until you explicitly take it off.

I made this cosmetic because that was recommended by one of the main devs. It doesn't matter to me either way.

File: Patches.py (hunk 1 above)
Comment (mracsys) [reply Patches.py]: Thanks, you're right, oversight on my part

**Comment by r0bd0g** [comment]
I mean, unequipping them when you load in was one suggestion. Another could be to not save that a mask is equipped. I don't think having to reequip it after a savewarp is a significant hit to the feature's convenience.

File: Patches.py (hunk 1 above)
Comment (mracsys) [reply Patches.py]: This was moved to the cosmetic function with new logic to add in the original table if the cosmetic is disabled.

**Comment by cjohnson57** [comment]
I also feel it shouldn't be cosmetic, gonna discuss with Cuphat and come back to you with a final answer

**Comment by cjohnson57** [comment]
Ok, decision is that it shouldn't be cosmetic

**Review by cjohnson57 — APPROVED** [review]
Thanks!

**Pull request merged**